lib/tasks/accounts.rake

```diff
@@ -2,43 +2,47 @@
 
 namespace :accounts do
   desc "Perform migration"
-  task :migration, %i[archive_path new_user_name] => :environment do |_t, args|
-    puts "Account migration is requested"
-    args = %i[archive_path new_user_name].map {|name| [name, args[name]] }.to_h
+  task :migration, %i[archive_path photos_path new_user_name] => :environment do |_t, args|
+    puts "Account migration is requested. You can import a profile or a photos archive or booth."
+    args = %i[archive_path photos_path new_user_name].map {|name| [name, args[name]] }.to_h
     process_arguments(args)
-
-    begin
-      service = MigrationService.new(args[:archive_path], args[:new_user_name])
-      service.validate
-      puts "Warnings:\n#{service.warnings.join("\n")}\n-----" if service.warnings.any?
-      if service.only_import?
-        puts "Warning: Archive owner is not fetchable. Proceeding with data import, but account migration record "\
-          "won't  be created"
-      end
-      print "Do you really want to execute the archive import? Note: this is irreversible! [y/N]: "
-      next unless $stdin.gets.strip.casecmp?("y")
-
-      start_time = Time.now.getlocal
-      service.perform!
-      puts service.only_import? ? "Data import complete!" : "Data import and migration complete!"
-      puts "Migration took #{Time.now.getlocal - start_time} seconds"
-    rescue MigrationService::ArchiveValidationFailed => exception
-      puts "Errors in the archive found:\n#{exception.message}\n-----"
-    rescue MigrationService::MigrationAlreadyExists
-      puts "Migration record already exists for the user, can't continue"
+    start_time = Time.now.getlocal
+    if args[:new_user_name].present?
+      import_profile = ImportProfileService.new
+      import_profile.import_by_files(args[:archive_path], args[:photos_path], args[:new_user_name])
+    else
+      puts "Must set a user name and a archive file path or photos file path"
     end
+    puts "\n Migration finished took #{Time.now.getlocal - start_time} seconds. (Photos might still be processed)"
   end
 
   def process_arguments(args)
-    if args[:archive_path].nil?
-      print "Enter the archive path: "
-      args[:archive_path] = $stdin.gets.strip
-    end
-    if args[:new_user_name].nil?
-      print "Enter the new user name: "
-      args[:new_user_name] = $stdin.gets.strip
-    end
+    request_archive_parameter(args)
+    request_photos_parameter(args)
+    request_username(args)
     puts "Archive path: #{args[:archive_path]}"
+    puts "Photos path: #{args[:photos_path]}"
     puts "New username: #{args[:new_user_name]}"
   end
 end
+
+def request_archive_parameter(args)
+  return unless args[:archive_path].nil?
```

Comment:
I would create a generic `request_parameter` method with a `arg` (`:archive_path`) and `text` (`"Enter the archive (.json, .gz, .zip) path: "`) parameter (in addition to the `args` parameter), because otherwise all these method are very similar.